client/src/components/Viewer/Viewer.js

```diff
@@ -37,169 +40,188 @@ function Viewer() {
-      // console.error('Connected:', socket.connected); // Should be true
-      setIsConnected(true);
-      inProgressUpload.current += 1;
-      socket.emit('addUser', {
-        token: JSON.parse(localStorage.getItem('dfs-user'))?.['token'],
-        inProgress: inProgressUpload.current,
-        socket_id: socket.id
+      socket.on('progress', progress_data => {
+        if (
+          progress_data.Data.Uploaded_Files !== undefined &&
+          progress_data.Data.Total_Files !== undefined
+        ) {
+          let num = progress_data.Data.Uploaded_Files;
+          let den = progress_data.Data.Total_Files;
+          let per = (num / den) * 100;
+          const fileName = file.name;
+          console.warn('per--', per);
+          setUploadPercentage(prevValue => ({
+            ...prevValue,
+            [fileName]: {
+              ...prevValue[fileName],
+              minio: per,
+            },
+          }));
+        }
+      });
+
+      socket.on('error', error => {
+        console.error('Socket.IO error:', error);
       });
-      const fileName = currentFile.name.name;
-      setUploadPercentage(prevValue => ({
-        ...prevValue,
-        [fileName]: {
-          minio: -1,
-          dzsave: -1,
-        },
-      }));
-    });
 
-    socket.on('progress', progress_data => {
-      if (
-        progress_data.Data.Uploaded_Files !== undefined &&
-        progress_data.Data.Total_Files !== undefined
-      ) {
-        let num = progress_data.Data.Uploaded_Files;
-        let den = progress_data.Data.Total_Files;
-        let per = (num / den) * 100;
-        const fileName = currentFile.name.name;
-        console.warn('per--', per);
+      socket.on('dzsave-progress', progress_data => {
+        let per = progress_data.progress;
+        const fileName = file.name;
         setUploadPercentage(prevValue => ({
           ...prevValue,
           [fileName]: {
             ...prevValue[fileName],
-            minio: per,
+            dzsave: Number(per),
           },
         }));
-      }
-    });
-
-    socket.on('error', error => {
-      console.error('Socket.IO error:', error);
-    });
-
-    socket.on('dzsave-progress', progress_data => {
-      let per = progress_data.progress;
-      const fileName = currentFile.name.name;
-      setUploadPercentage(prevValue => ({
-        ...prevValue,
-        [fileName]: {
-          ...prevValue[fileName],
-          dzsave: Number(per),
-        },
-      }));
-    });
+      });
 
-    socket.on('AddedUser', async () => {
-      const formData = new FormData();
-      setDisplayProgressBar(true);
-      formData.append('file', currentFile.name);
-      let bucketURL = config.BASE_URL + '/objects/' + shortEmail;
+      socket.on('AddedUser', async () => {
+        const formData = new FormData();
+        setDisplayProgressBar(true);
+        formData.append('file', file);
+        let bucketURL = config.BASE_URL + '/objects/' + shortEmail;
 
```

Comment:
what is this socket message used for?

Reply:
this mssg is sent from the server to the client to tell that the user has been added and thus the API request can now be sent. Thus it establishes a flow like socket connect -> user added -> API request